### PR TITLE
 Optimization error in update operation

### PR DESCRIPTION
**Pull request by anthony-redFox**

### CODE REVIEW

File: src/update.js [hunk 1]
```diff
@@ -23,6 +23,10 @@ function err(operation, expectedTarget, path) {
   );
 }
 
+const tr = true;
+const arrayMethods = {push: tr, unshift: tr, concat: tr, splice: tr, pop: tr, shift: tr};
```
Comment (anthony-redFox) [inline src/update.js]: Will be nice to use Set.
@Yomguithereal  what do you think about drop IE9 and IE10?

**Comment by Zache** [comment]
Please don't

Den 29 dec. 2017 21:50 skrev "Anton" <notifications@github.com>:

**Comment by anthony-redFox** [comment]
@Zache please don't, what?

**Comment by Zache** [comment]
Drop support for IE9 and IE10

Den 30 dec. 2017 08:56 skrev "Anton" <notifications@github.com>:

**Comment by anthony-redFox** [comment]
@Zache  Why do I ask? 
1. Microsoft doesn't support [IE9 and IE10 at all](https://www.microsoft.com/en-us/WindowsForBusiness/End-of-IE-support)
2. Most libraries like [lodash](https://lodash.com), [mocha](https://github.com/mochajs/mocha/issues/3148) already drop or will drop IE9 and IE10.
3. I am working on enterprise project (PLM system for big companies) and we don't support IE9 and IE10

Can you explain, why IE9 or IE10 still work for you?

**Comment by Yomguithereal** [comment]
Hello @anthony-redFox. I don't really see the point of your PR. What are you trying to fix or enhance here?

**Comment by anthony-redFox** [comment]
Hello, @Yomguithereal  it is just optimization of throw a error and as result the bundle will be less size.

**Comment by Yomguithereal** [comment]
@anthony-redFox. Your edit should produce more checks and conditions than before. I am not sure it improves performance. Since you test the existence of a string as an object key now.

**Comment by anthony-redFox** [comment]
I agree with you that I added additional check.

Also by benchmark nothing changed
Without patch:

Baobab#get x 589,771 ops/sec ±0.76% (94 runs sampled)
Baobab#set x 185,282 ops/sec ±0.31% (97 runs sampled)
Baobab#unset x 849,941 ops/sec ±0.75% (95 runs sampled)
Baobab.Cursor#set x 38,957 ops/sec ±4.22% (89 runs sampled)
Baobab.Cursor#unset x 815,053 ops/sec ±0.36% (93 runs sampled)
Baobab.Cursor#push x 1,082 ops/sec ±44.72% (10 runs sampled)
Baobab.Cursor#unshift x 464 ops/sec ±1.39% (84 runs sampled)
Baobab.Cursor#splice x 538 ops/sec ±1.73% (89 runs sampled)

With patch

Baobab#get x 595,541 ops/sec ±0.73% (95 runs sampled)
Baobab#set x 183,132 ops/sec ±0.46% (94 runs sampled)
Baobab#unset x 833,419 ops/sec ±2.08% (91 runs sampled)
Baobab.Cursor#set x 38,183 ops/sec ±4.43% (89 runs sampled)
Baobab.Cursor#unset x 829,665 ops/sec ±0.51% (95 runs sampled)
Baobab.Cursor#push x 1,034 ops/sec ±40.79% (10 runs sampled)
Baobab.Cursor#unshift x 474 ops/sec ±1.36% (82 runs sampled)
Baobab.Cursor#splice x 557 ops/sec ±0.34% (93 runs sampled)

If you still disagree with me, please close PR.

**Comment by Yomguithereal** [comment]
Difference seems negligible and could be due to other factors affecting the engine outside of our control. No?

**Comment by anthony-redFox** [comment]
Yes, right.

**Comment by Yomguithereal** [comment]
Ok. I will close this then. Thanks for the contribution anyway @anthony-redFox.

**Pull request closed without merging**